### ISSUE TITLE
Android - General - TalkBack Foucs Ring Does Not have Enough Color Contrast

### ISSUE THREAD

**Issue by jfc3-dol** [USDepartmentofLabor/Comply-Chain#7]
When reading a screen with TalkBack, I’m having trouble noticing the light blue focus ring when it gets to the “Menu” button that is also blue. So my suggestion is to change the focus ring to black ( #000000 ), so it can be seen and has enough color contrast. The problem shows up when reading through the items on the home page screen too. Will be creating an issue for that problem.

Focus ring is tiny and thin so there is now screenshot.

**Comment by KyleLivingstonDOL** [comment]
Thanks John. We will work on this.

**Comment by jfc3-dol** [comment]
So keyboard users or those using a screen reader to read along can see where on a page/screen they are located.

Yes, the outline ring needs to have enough color contrast between itself and the background color along with itself, and in this example, the button. In my years of programming, I have found that inverting the color of the button and text color can work the best. An example would be if the text is white and the background of the button is dark blue when it in its normal state that if you make the text and border color the same blue with a white background, you should be fine unless the button is on top of another background color that won't work.

I use Jonathan Snook's online Colour Contrast Checker ( https://snook.ca/technical/colour_contrast/colour.html ) to make sure the color contrast ratio is 4.5:1, and both of the WCAG 2.0 AA boxes say "YES". All you have to do is put the text and background colors in the foreground and background fields, and it will tell you what you need to know.

**Comment by KyleLivingstonDOL** [comment]
It looks like this has been fixed. UAT for Sprint 7 (which is ongoing right now) will start on June 10.

**Comment by LeHaine** [comment]
https://github.com/USDepartmentofLabor/Comply-Chain/issues/37#issuecomment-498685874

**Comment by jfc3-dol** [comment]
The TalkBack blue focus ring does not have enough color contrast when focusing on the blue or dark blue buttons.

If this is an Android operating system issue I would like to keep this open until there is a fix since you said CSS changes aren't working.

**Comment by KyleLivingstonDOL** [comment]
Zolon, please let us know when the CSS changes will start working again.

**Comment by KyleLivingstonDOL** [comment]
Update: This relates to #18 and #37; therefore this should be on hold too.

**Comment by jfc3-dol** [comment]
If that is the case with this issue or any other that is not left open and not accessible OPA sends out a document listing the number of issues found, still open, and requests at least an e-mail stating the given agency is willing to take the risk if someone complains in whatever way that it's not accessible. Which has been done by all of them as far as I know.

**Comment by KyleLivingstonDOL** [comment]
Ok. Thanks John. I appreciate this information. Do you know what the procedure is to start this process?

**Comment by jfc3-dol** [comment]
Once we are done testing the entire Comply Chain application (iOS, Android, and web) I create and send to you and my boss OPAs findings of the number of issues found; the number closed, etc. At which point we then look for a signature (can be electronic or an e-mail) from the agency that they are willing to take the risk of having known accessibility issues when moving everything to PROD.

By documenting things in GitHub with all these types of comments, we could explain to someone that some of them are platform issues, etc. OPA doesn't want to put an agency or DOL at risk for complaints, etc. from people along with we are working on them.

**Comment by jfc3-dol** [comment]
Marking this issue for a "Future release" by using the label "future release" so we all know we are most likely not finishing it this release unless we manage to figure out how to do it.

**Comment by KyleLivingstonDOL** [comment]
Thanks John.

**Comment by jfc3-dol** [comment]
The TalkBack focus ring still does not have enough color contrast and might be the default from Android, so it's possibly not fixable.

**Comment by pbhatt17** [comment]
@jfc3-dol 
This ring is from talkback and can't modify. Do you still want to keep this ticket open?

**Comment by jfc3-dol** [comment]
@pbhatt17 

I'm keeping this open to remind us that the TalkBack focus ring still has poor color contrast. It probably won't close until Android allows developers to change the color of the focus ring when using TalkBack or a blue tooth keyboard.
